### PR TITLE
chore(suite): refactoring: split components for DeviceItem

### PR DESCRIPTION
**Pull request by peter-sanderson**
Just another refactoring to split some spaghetti code.

\+ Better usage of variants for Text and Icon component

image

### CODE REVIEW

File: packages/components/src/components/typography/Text/Text.tsx [hunk 1]
```diff
@@ -3,13 +3,16 @@ import { UIVariant } from '../../../config/types';
 import { CSSColor, Color, Colors, TypographyStyle, typography } from '@trezor/theme';
 import { ReactNode } from 'react';
 
-export type TextVariant = Extract<UIVariant, 'primary' | 'info' | 'warning' | 'destructive'>;
+export type TextVariant =
+    | Extract<UIVariant, 'primary' | 'info' | 'warning' | 'destructive'>
+    | 'subdued';
```
Comment (peter-sanderson) [inline packages/components/src/components/typography/Text/Text.tsx]: This is controversial, @jvaclavik lets discuss this separately.

File: packages/components/src/components/typography/Text/Text.tsx [hunk 2]
```diff
@@ -18,7 +21,11 @@ type ColorProps = {
     $color?: string;
 };
 
-const getColor = ({ $variant, theme, $color }: ColorProps): CSSColor | 'inherit' | string => {
+export const getColorForTextVariant = ({
```
Comment (peter-sanderson) [inline packages/components/src/components/typography/Text/Text.tsx]: exporting this is also a bit controversial, @jvaclavik we may need to discuss ho to work with this and what is proper place for this function

**Review by komret — APPROVED** [review]

**Comment by komret** [comment]
@jvaclavik please check the controversial stuff.

File: packages/components/src/components/assets/Icon/Icon.tsx [hunk 3]
```diff
@@ -3,15 +3,51 @@ import styled, { css, useTheme } from 'styled-components';
...
+
+    return $variant === undefined ? theme.iconDefault : theme[variantColorMap[$variant]];
+};
 
 export type IconType = keyof typeof ICONS;
 
-const SvgWrapper = styled.div<{
-    $color: WrapperProps['color'];
+type SvgWrapperProps = {
+    $color: WrapperProps['color'] | undefined;
```
Comment (jvaclavik) [inline packages/components/src/components/assets/Icon/Icon.tsx]: why not `$color?: WrapperProps['color']`

File: packages/suite/src/views/suite/SwitchDevice/DeviceItem/DeviceDetail.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    align-self: center;
+`;
+
+const DeviceLabel = styled.div`
+    ${typography.body};
+    margin-bottom: -${spacingsPx.xxs};
+    min-width: 0;
+    color: ${({ theme }) => theme.textDefault};
+    overflow: hidden;
+    text-overflow: ellipsis;
```
Comment (jvaclavik) [inline packages/suite/src/views/suite/SwitchDevice/DeviceItem/DeviceDetail.tsx]: You might use `TruncateWithTooltip` here

**Review by jvaclavik — APPROVED** [review]

File: packages/components/src/components/assets/Icon/Icon.tsx (hunk 3 above)
Comment (peter-sanderson) [reply packages/components/src/components/assets/Icon/Icon.tsx]: To make it required so you dont forget to pass it.

Comment (jvaclavik) [reply packages/components/src/components/assets/Icon/Icon.tsx]: Why should we pass it even though it's undefined?

Comment (peter-sanderson) [reply packages/components/src/components/assets/Icon/Icon.tsx]: Its a consequence of using this trick

image

File: packages/suite/src/views/suite/SwitchDevice/DeviceItem/DeviceDetail.tsx (hunk 4 above)
Comment (peter-sanderson) [reply packages/suite/src/views/suite/SwitchDevice/DeviceItem/DeviceDetail.tsx]: :heavy_check_mark:

**Pull request merged**